hrabsence/CRM/HRAbsence/Form/AbsenceType.php

```diff
@@ -106,11 +106,32 @@ public function buildQuickForm() {
     $this->addFormRule(array('CRM_HRAbsence_Form_AbsenceType', 'formRule'), $this);
   }
 
-  static function formRule($fields, $files, $self) {
-    $errors = array();
+  /**
+   * Used in validation from addFormRule
+   *
+   * @param array $fields
+   *  An array of fields from the form
+   * @param array $files
+   *  An array of uploaded files
+   * @param self $self
+   *  An instance of this class
+   * @return array
+   *   An array of errors
+   */
+  public static function formRule($fields, $files, $self) {
```

Comment:
If you need $this inside formRule, do this
- Change the method to be an instance method instead of static
- On line 106, change `array('CRM_HRAbsence_Form_AbsenceType', 'formRule')` to `array($this, 'formRule')`

and then you can use `$this` and get rid of `$self`